### PR TITLE
Minor docs correction for maxintfloat.

### LINKED ISSUE #29297: maxintfloat should return an ::Integer
Contrary to its [documentation](https://github.com/JuliaLang/julia/blob/ce6454642ca2f6075623179c6804ea6c78d1fdb5/base/floatfuncs.jl#L25) which says (emphasis mine)

> ... `maxintfloat` returns the smallest positive **integer** ...

it actually returns floating point values.

I don't know if this is a problem with wording or the implementation. IMO returning integers would be more useful as one could perform exact arithmetic & comparisons with them around the edge of the representable region.

### PR DESCRIPTION
**Pull request by tpapp**
Fixes #29297, see discussion there.

### CODE REVIEW

File: base/floatfuncs.jl [hunk 1]
```diff
@@ -19,11 +19,13 @@ signbit(x::Float16) = signbit(bitcast(Int16, x))
...
-(which defaults to `Float64`).
+The largest consecutive integer-valued floating point number that is exactly represented in
+the given floating-point type `T` (which defaults to `Float64`).
 
-That is, `maxintfloat` returns the smallest positive integer `n` such that `n+1`
-is *not* exactly representable in the type `T`.
+That is, `maxintfloat` returns the smallest positive integer-valued floating point number
+`n` such that `n+1` is *not* exactly representable in the type `T`.
+
+When an integer is needed, use `Integer(maxintfloat(T))`.
```
Comment (stevengj) [inline base/floatfuncs.jl]: When an integer type is needed

Comment (tpapp) [reply base/floatfuncs.jl]: I am not sure I understand, as `Integer(maxintfloat(T))` will return an integer (value), not a type.

Comment (StefanKarpinski) [reply base/floatfuncs.jl]: “Integer-typed value”?

Comment (stevengj) [reply base/floatfuncs.jl]: The point is that `maxintfloat(T)` returns an integer already, so your current wording is misleading.

"When an `Integer`-type value is needed" seems okay.

File: base/floatfuncs.jl [hunk 2]
```diff
@@ -19,11 +19,13 @@ signbit(x::Float16) = signbit(bitcast(Int16, x))
 """
     maxintfloat(T=Float64)
 
-The largest consecutive integer that is exactly represented in the given floating-point type `T`
-(which defaults to `Float64`).
+The largest consecutive integer-valued floating point number that is exactly represented in
```
Comment (stevengj) [inline base/floatfuncs.jl]: "floating-point number" should be hyphenated (compound adjective)

**Review by stevengj — APPROVED** [review]

**Pull request merged**